### ISSUE TITLE
`git.Clone()` throwing "Early EOL" error

### ISSUE THREAD

**Issue by callumacrae** [nodegit/nodegit#555]
Hey!

These three lines of code result in an error saying only "Early EOL" being thrown: https://github.com/Lostmyname/parker-pr-bot/blob/master/src/get-file.js#L28-L30

The code works locally on my machine, but not on the server I'm trying to deploy my app to. The `git` cli works, though, and my SSH key is correct.

Any ideas? I'm on gitter if you want to talk about it here, and I'll document it here so that it's searchable.

**Comment by bso-intel** [comment]
I got the same issue "undefined is not a function" error is thrown from the git.Clone() call.

**Comment by callumacrae** [comment]
I think that's probably a completely different issue. Could you open another issue and post your code in it? :smiley:

**Comment by bso-intel** [comment]
Sure. Sorry for the noise.

**Comment by tbranyen** [comment]
Can you try with NodeGit 0.4.0 (in NPM).  Thanks!

**Comment by callumacrae** [comment]
Nope, still got the same error. Again, only happening on my server running Ubuntu—could be a Linux problem? I'm running OS X locally.

```
  1) get-file() should clone and get file:
     Error: Early EOF
      at Error (native)

  2) get-file() should get file from other branch:
     Error: Early EOF
      at Error (native)

  3) get-file() should ensure build commands have ran:
     Error: Early EOF
      at Error (native)
```

**Comment by maxkorp** [comment]
Just out of curiosity, do all of your tracked files have an ending newline? I'm wondering if there's something unusual regarding that in libgit2 that works in true git.

**Comment by callumacrae** [comment]
It's the same both locally and remotely: `git@github.com:Lostmyname/eagle.git`. It's a private repo, but if git has access, I don't see why libgit2 shouldn't.

`console.log(e, e.code)` outputs `[Error: Early EOF] undefined`.

**Comment by maxkorp** [comment]
And you're sure that your ssh-agent has the key loaded? Try running `ssh-add` (assuming you're using `~/.ssh/id_rsa`) beforehand. I know I've experienced git loading the key but libgit2 not. I think git might default to checking id_rsa in some cases, whereas libgit2/nodegit are just going to check the agent and thats it.

**Comment by maxkorp** [comment]
@callumacrae Can you try yet again with 0.4.1?
